stub.sh

```diff
@@ -366,7 +389,9 @@ __stub_call() {
 
   local index="$(__stub_index "$cmd")"
   if [ -n "$index" ]; then
-    eval "STUB_${index}_CALLS+=(\"\$args\")"
+    # Send call to the main process
+    echo "STUB_${index}_CALLS+=(\"$args\")" > $STUB_FIFO &
+    kill -USR1 $MAIN_SPID
```

Comment:
I'm a little worried here about race conditions. I could not came up with any solutions though. Haven't seen any issues with the test cases though